frontend/src/routes/Infrastructure/InfraEnvironments/Details/InfraEnvironmentDetailsPage.tsx

```diff
@@ -184,6 +237,20 @@ const InfraEnvironmentDetailsPage: React.FC<InfraEnvironmentDetailsPageProps> =
                 usedHostnames={usedHostnames}
                 isBMPlatform={isBMPlatform(infrastructures[0])}
             />
+            <AddBmcHostModal
+                infraEnv={infraEnv}
+                isOpen={isBmcModalOpen}
+                onClose={() => setBMCModalOpen(false)}
+                onCreateBMH={getOnCreateBMH(infraEnv)}
+                onSaveISOParams={getOnSaveISOParams(infraEnv)}
+                usedHostnames={usedHostnames}
+                isBMPlatform={isBMPlatform(infrastructures[0])}
+            />
+            <AddBmcHostYamlModal
+                isOpen={isBmcYamlModalOpen}
+                onClose={() => setBMCYamlModalOpen(false)}
+                onCreateBmcByYaml={importYaml}
+            />
```

Comment:
its better to not render the modal at all, when its closed
```suggestion
            {isBmcModalOpen && <AddBmcHostModal
                infraEnv={infraEnv}
                isOpen={isBmcModalOpen}
                onClose={() => setBMCModalOpen(false)}
                onCreateBMH={getOnCreateBMH(infraEnv)}
                onSaveISOParams={getOnSaveISOParams(infraEnv)}
                usedHostnames={usedHostnames}
                isBMPlatform={isBMPlatform(infrastructures[0])}
            />
            }
            {isBmcYamlModalOpen && <AddBmcHostYamlModal
                isOpen={isBmcYamlModalOpen}
                onClose={() => setBMCYamlModalOpen(false)}
                onCreateBmcByYaml={importYaml}
            />}
```